src/connector/src/parser/unified/json.rs

```diff
@@ -0,0 +1,446 @@
+            (Some(DataType::Boolean) | None, ValueType::Bool) => value.as_bool().unwrap().into(),
+
+            (
+                Some(DataType::Boolean),
+                ValueType::I64 | ValueType::I128 | ValueType::U64 | ValueType::U128,
+            ) if matches!(self.boolean_handing, BooleanHandling::Relax { .. })
+                && matches!(value.as_i64(), Some(0i64) | Some(1i64)) =>
+            {
+                (value.as_i64() == Some(1i64)).into()
+            }
+
+            (Some(DataType::Boolean), ValueType::String)
+                if matches!(
+                    self.boolean_handing,
+                    BooleanHandling::Relax {
+                        string_parsing: true,
+                        ..
+                    }
+                ) =>
+            {
+                match value.as_str().unwrap().to_lowercase().as_str() {
+                    "true" => true.into(),
+                    "false" => false.into(),
+                    "1" if matches!(
+                        self.boolean_handing,
+                        BooleanHandling::Relax {
+                            string_parsing: true,
+                            string_integer_parsing: true
+                        }
+                    ) =>
+                    {
+                        true.into()
+                    }
+                    _ => Err(create_error())?,
+                }
+            }
+            // ---- Int16 -----
+            (
+                Some(DataType::Int16),
+                ValueType::I64 | ValueType::I128 | ValueType::U64 | ValueType::U128,
+            ) => value.try_as_i16()?.into(),
+
+            (Some(DataType::Int16), ValueType::String)
+                if matches!(
+                    self.numeric_handling,
+                    NumericHandling::Relax {
+                        string_parsing: true
+                    }
+                ) =>
+            {
+                value
+                    .as_str()
+                    .unwrap()
+                    .parse::<i16>()
+                    .map_err(|_| create_error())?
+                    .into()
+            }
+            // ---- Int32 -----
+            (
+                Some(DataType::Int32),
+                ValueType::I64 | ValueType::I128 | ValueType::U64 | ValueType::U128,
+            ) => value.try_as_i32()?.into(),
+
+            (Some(DataType::Int32), ValueType::String)
+                if matches!(
+                    self.numeric_handling,
+                    NumericHandling::Relax {
+                        string_parsing: true
+                    }
+                ) =>
+            {
+                value
+                    .as_str()
+                    .unwrap()
+                    .parse::<i32>()
+                    .map_err(|_| create_error())?
+                    .into()
+            }
+            // ---- Int64 -----
+            (None, ValueType::I64 | ValueType::U64) => value.try_as_i64()?.into(),
```

Comment:
in which case will expect type be None

Reply:
The `type_expected` is an Option. If None is provided, it will infer the target value type by JsonValue.